src/frontend/packages/core/src/shared/components/list/list-types/cf-users/cf-user-list-config.service.ts

```diff
@@ -109,6 +111,50 @@ export class CfUserListConfigService extends ListConfig<APIResource<CfUser>> {
     description: `Manage roles`,
   };
 
+  removeUserActions(): IListAction<APIResource<CfUser>>[] {
+    const activeRouteCfOrgSpace = this.cfUserService.activeRouteCfOrgSpace;
+    const orgGuid = activeRouteCfOrgSpace.orgGuid;
+    const spaceGuid = activeRouteCfOrgSpace.spaceGuid;
+    const action = (withSpaces?: boolean) => {
+      return (user: APIResource<CfUser>) => {
+        const queryParams = { queryParams: { user: user.entity.guid, spaces: withSpaces } };
+
+        this.store.dispatch(new UsersRolesSetUsers(activeRouteCfOrgSpace.cfGuid, [user.entity]));
+        this.router.navigate([this.createRemoveUserUrl()], queryParams);
+      };
+    };
+    const createVisible = (userRow$: Observable<APIResource<CfUser>>) => combineLatest(
+      userRow$,
+      this.createCanUpdateOrgSpaceRoles()
+    ).pipe(
+      map(([user, canUpdateRoles]) => {
+        const isCfContext = !orgGuid && !spaceGuid;
+        const hasRoleWithinOrgsSpaces = hasRoleWithin(user.entity, orgGuid, spaceGuid) ||
```

Comment:
Had a quick try of this and the 'remove from space' action was showing when the user had no space roles. We should also think if there's a neat way to alter the 'remove from org and space' text as well

Reply:
Had another look today and noticed the following
- connected user is non cf admin, `user` of an org, `manager` of space1 in that org and `developer` of space2 in that org
- at the org level user list both 'remove' action menu items are shown.
- If they're not an org manager of that org then the `remove from orgs and spaces` should not be shown
- If they're not an org manager and only space manager of some then either
  - hide `remove from spaces` 
  - show `remove from spaces` but only include those that they're a space manager of
- There's an interested cf related bug to this, will update on our next sync

Reply:
I think there's only one thing left, plus the e2e secrets.yaml change, before this can go in and that's the second from last point above. I think, given how difficult determining the user's manager role for each space would be tricky, so lets go with...
- Show the `remove from spaces` option as currently
- When displaying/determining role changes in the stepper we check that the user has the right to modify that org/space
- If, after removing the roles as per above, there are none left show a suitable message on screen and disable the `apply` button

Reply:
I've changed some bits that were wrong. Take a look at it again and lemme know if it's working as expected or not.

Reply:
I still see the same issue. Spaces where the connected user is not a manager (or manager of the org) are shown after clicking on the 'remove from spaces' button.